backend/core/src/main/java/org/sonarsource/sonarlint/core/embedded/server/ShowIssueRequestHandler.java

```diff
@@ -184,23 +190,31 @@ public static class ShowIssueQuery {
     private final String tokenName;
     @Nullable
     private final String tokenValue;
+    @Nullable
+    private final String organizationKey;
 
     public ShowIssueQuery(String serverUrl, String projectKey, String issueKey, String branch,
-      @Nullable String pullRequest, @Nullable String tokenName, @Nullable String tokenValue) {
+      @Nullable String pullRequest, @Nullable String tokenName, @Nullable String tokenValue, @Nullable String organizationKey) {
       this.serverUrl = serverUrl;
       this.projectKey = projectKey;
       this.issueKey = issueKey;
       this.branch = branch;
       this.pullRequest = pullRequest;
       this.tokenName = tokenName;
       this.tokenValue = tokenValue;
+      this.organizationKey = organizationKey;
     }
 
     public boolean isValid() {
       return isNotBlank(serverUrl) && isNotBlank(projectKey) && isNotBlank(issueKey) && isNotBlank(branch)
+        && (!isSonarCloud(serverUrl) || isNotBlank(organizationKey))
```

Comment:
Do you also send the `serverUrl` in the case of SonarCloud as part of the request? Is it needed?

Reply:
We need the server url just to check whether SC or not. We can also use the origin header. I'll use the origin header then